src/controllers/controllerinputmappingtablemodel.cpp

```diff
@@ -222,14 +212,24 @@ QVariant ControllerInputMappingTableModel::data(const QModelIndex& index,
                 }
                 return QVariant::fromValue(mapping.options);
             case MIDI_COLUMN_ACTION: {
+                const auto* const pControl = std::get_if<ConfigKey>(&mapping.control);
+                if (!pControl) {
+                    const auto* const ppJSValue =
+                            std::get_if<std::shared_ptr<QJSValue>>(
+                                    &mapping.control);
+                    if (ppJSValue) {
```

Comment:
inner shared_ptr could be null (at least judging by type system).
```suggestion
                    if (ppJSValue && *ppJSValue) {